sisl/geom/category/_coord.py

```diff
@@ -295,15 +285,18 @@ def _new(cls, *interval, **kwargs):
         """
         def _apply_key(k, v):
             return f"{key}_{k}", v
-        new_kwargs = dict(map(_apply_key, *zip(*kwargs.items())))
+        if len(kwargs) > 0:
+            new_kwargs = dict(map(_apply_key, *zip(*kwargs.items())))
+        else:
+            new_kwargs = {}
 
         # Convert interval to correct interpretation
         if len(interval) > 1:
             raise ValueError(f"{cls.__name__} makes only sense to instantiate with 1 non-keyword argument")
```

Comment:
Actually, I was supporting `AtomX((1, 3))` because that's how you can call it from the kw builder (e.g.: `geom.sub({"x": (1,3)})`). You don't want to support this?

Reply:
I do want to support it, I can see I messed that up, I'll correct.

Reply:
fixed

Reply:
My mistake, I read it wrong :)  But I don't understand what do you expect `interval` to be. Because if `len(interval) == 1`, `ìnterval` is probably something like `((-2, 3), )`

Reply:
Yeah, the point is that one should not be able to call `AtomX(4.0, 3.0)` (this would explicitly fail), nor `AtomX(3.)` (this would fail in `AtomXYZ.__init__` due to ambiguity). It cannot be deferred. So only one interval is allowed.

Reply:
I expect `interval` to be a single tuple, `isinstance(interval[0], (list, tuple, np.ndarray))` and nothing more, otherwise I can't know what it means?

Reply:
> My mistake, I read it wrong :) But I don't understand what do you expect interval to be. Because if len(interval) == 1, ìnterval is probably something like ((-2, 3), )

Sorry, your comment didn't show up, that's why I commented this.

I used `*interval` so that when you did `AtomX(3, 4)`, `interval` was `(3,4)`, but if you don't want to support it (that's what I understand when you raise the error on `len(interval) > 1`), you can just have `interval=None` as the first argument and that will save us some checks right?

Reply:
No, that is a good idea, see commit, fixed.